File: py/2017/01-InverseCaptcha/solution.py
import os

with open("input.txt", "r") as file:
	input = file.read()

def solution_part1():
	solution = 0
	for i in range(0, len(input)):
		char = input[i]
		next = input[(i + 1) % len(input)]
		if char is next:
			solution += int(char)
	return solution

def solution_part2():
	solution = 0
	for i in range(0, len(input)):
		char = input[i]
		next = input[int(i + len(input) / 2) % len(input)]
		if char is next:
			solution += int(char)
	return solution

print("Solution Part 1:", solution_part1())
print("Solution Part 2:", solution_part2())

File: cpp/2017/01-InverseCaptcha/solution.cpp
#include <iostream>
#include <fstream>

#define INPUT_BUFFER_SIZE 2134

int solution_part1(char (&input)[INPUT_BUFFER_SIZE]) {
	int solution = 0;
	for (int i = 0; i < INPUT_BUFFER_SIZE; i++) {
		char& byte = input[i];
		char& next = input[(i + 1) % INPUT_BUFFER_SIZE];
		if (byte == next) {
			solution += byte - 48;
		}
	}
	return solution;
}

int solution_part2(char (&input)[INPUT_BUFFER_SIZE]) {
	int solution = 0;
	for (int i = 0; i < INPUT_BUFFER_SIZE; i++) {
		char& byte = input[i];
		char& next = input[(i + (INPUT_BUFFER_SIZE / 2)) % INPUT_BUFFER_SIZE];
		if (byte == next) {
			solution += byte - 48;
		}
	}
	return solution;
}

int main() {
	std::ifstream file("input.txt");
	char buffer[INPUT_BUFFER_SIZE];
	file.read(buffer, INPUT_BUFFER_SIZE);
	file.close();

	std::cout << "Solution Part 1: " << solution_part1(buffer) << "\n";
	std::cout << "Solution Part 2: " << solution_part2(buffer) << "\n";
}